service/collector/app_resource.go

```diff
@@ -79,24 +113,38 @@ func (c *AppResource) collectAppStatus(ctx context.Context, ch chan<- prometheus
 			app.Status.Version,
 		)
 
-	}
+		if !key.IsCordoned(app) {
+			continue
+		}
 
-}
+		t, err := convertToTime(key.CordonUntil(app))
+		if err != nil {
+			c.logger.Log("level", "warning", "message", "could not convert cordon-until", "stack", fmt.Sprintf("%#v", err))
+			continue
+		}
 
-// Collect is the main metrics collection function.
-func (c *AppResource) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
+		ch <- prometheus.MustNewConstMetric(
+			appCordonExpireTimeDesc,
+			prometheus.GaugeValue,
+			float64(t.Unix()),
+			key.AppName(app),
+		)
+	}
+	return nil
+}
 
-	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting metrics")
+func convertToTime(datetime string) (time.Time, error) {
```

Comment:
```suggestion
func convertToTime(input string) (time.Time, error) {
```

Nitpick but I find the var name also being a "type" a bit confusing.